src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,38 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+interface Props {
+  onResetSelectedTodo: () => void;
+  selectedTodo: Todo;
+}
+
+export const TodoModal: React.FC<Props> = ({
+  onResetSelectedTodo,
+  selectedTodo,
+}) => {
+  const [loadingTodoModal, setLoadingTodoModal] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    setLoadingTodoModal(true);
+    getUser(selectedTodo.userId)
+      .then(setUser)
+      .finally(() => setLoadingTodoModal(false));
+  }, [selectedTodo.userId]);
+
+  const handleResetUserAndTodo = () => {
+    setUser(null);
+    onResetSelectedTodo();
+  };
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {loadingTodoModal ? (
```

Comment:
You are generating the key on render here. This can lead to performance issues because react uses the key to identify elements and if the key changes on every render, react will think it's a new element and will re-render it. Instead, you should generate the key when the data is created or received.